plotly/tools.py

```
@@ -2274,6 +2314,160 @@ def create_candlestick(open, high, low, close,
+
+        rug_text = ['a', 'b', 'c', 'd', 'e',
+                    'f', 'g', 'h', 'i', 'j',
+                    'k', 'l', 'm', 'n', 'o']
+
+        # Create distplot
+        fig = FF.create_distplot(
+            hist_data, group_labels, rug_text=rug_text, bin_size=.2)
+
+        # Add title
+        fig['layout'].update(title='Dist Plot')
+
+        # Plot!
+        url = py.plot(fig, filename='Distplot with rug text', validate=False)
+        ```
+
+        Example 3: Plot with normal curve and hide rug plot
+        ```
+        import plotly.plotly as py
+        from plotly.tools import FigureFactory as FF
+
+        x1 = np.random.randn(20)
+        x2 = np.random.randn(40)+1
+        x3 = np.random.randn(10)-1
+        x4 = np.random.randn(10)+2
+
+        hist_data = [x1] + [x2] + [x3] + [x4]
+        group_labels = ['2012', '2013', '2014', '2015']
+
+        fig = FF.create_distplot(
+            hist_data, group_labels, curve_type='normal',
+            show_rug=False, bin_size=.4)
+
+        url = py.plot(fig, filename='hist and normal curve', validate=False)
+        """
+        FigureFactory.validate_distplot(hist_data, curve_type)
+        FigureFactory.validate_equal_length(hist_data, group_labels)
+
+        hist = _Distplot(
+            hist_data, group_labels, bin_size,
+            curve_type, **kwargs).make_hist()
+        if curve_type == 'normal':
+            curve = _Distplot(
+                hist_data, group_labels, bin_size,
+                curve_type, **kwargs).make_normal()
+        else:
+            curve = _Distplot(
+                hist_data, group_labels, bin_size,
+                curve_type, **kwargs).make_kde()
+        rug = _Distplot(
+            hist_data, group_labels, bin_size,
+            curve_type, **kwargs).make_rug()
+
+        data = []
+        if show_hist:
+            data.append(hist)
+        if show_curve:
+            data.append(curve)
+        if show_rug:
+            data.append(rug)
```

would it be cool to add all this data by default, but hide it in the figure? then they could always turn it on in the gui later?

@chriddyp thoughts?